guillotina/contrib/dbusers/content/users.py

```diff
@@ -0,0 +1,111 @@
+# -*- encoding: utf-8 -*-
+from guillotina import configure
+from guillotina import schema
+from guillotina.content import Folder
+from guillotina.contrib.dbusers import _
+from guillotina.directives import read_permission
+from guillotina.interfaces import Allow
+from guillotina.interfaces import IFolder
+from guillotina.interfaces import IPrincipal
+
+import typing
+
+
+class IUserManager(IFolder):
+    pass
+
+
+class IUser(IFolder, IPrincipal):
+
+    username = schema.TextLine(
+        title=_('Username'),
+        required=False)
+
+    email = schema.TextLine(
+        title=_('Email'),
+        required=False)
+
+    name = schema.TextLine(
+        title=_('Name'),
+        required=False)
+
+    read_permission(password='guillotina.Nobody')
+    password = schema.TextLine(
+        title=_('Password'),
+        required=False)
+
+    user_groups = schema.List(
+        title=_('Groups'),
+        value_type=schema.TextLine(),
+        required=False
+    )
+
+    user_roles = schema.List(
```

Comment:
I think this should also be write_protected so that a user cannot modify its own roles

Reply:
Agree!